core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Team.hpp

```diff
@@ -0,0 +1,393 @@
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_OPENACC_PARALLEL_REDUCE_TEAM_HPP
+#define KOKKOS_OPENACC_PARALLEL_REDUCE_TEAM_HPP
+
+#include <OpenACC/Kokkos_OpenACC_Team.hpp>
+#include <OpenACC/Kokkos_OpenACC_FunctorAdapter.hpp>
+
+#ifdef KOKKOS_ENABLE_OPENACC_COLLAPSE_HIERARCHICAL_CONSTRUCTS
+
+//----------------------------------------------------------------------------
+//----------------------------------------------------------------------------
+// Hierarchical Parallelism -> Team level implementation
+template <class FunctorType, class ReducerType, class... Properties>
+class Kokkos::Impl::ParallelReduce<FunctorType,
+                                   Kokkos::TeamPolicy<Properties...>,
+                                   ReducerType, Kokkos::Experimental::OpenACC> {
+ private:
+  using Policy = Kokkos::Impl::TeamPolicyInternal<Kokkos::Experimental::OpenACC,
+                                                  Properties...>;
+
+  using Analysis =
+      FunctorAnalysis<FunctorPatternInterface::REDUCE, Policy, FunctorType>;
+  using value_type   = typename Analysis::value_type;
+  using pointer_type = typename Analysis::pointer_type;
+
+  Kokkos::Experimental::Impl::FunctorAdapter<FunctorType, Policy> m_functor;
+  const Policy m_policy;
+  const ReducerType m_reducer;
+  const pointer_type m_result_ptr;
+
+ public:
+  inline void execute() const {
+    auto league_size   = m_policy.league_size();
+    auto team_size     = m_policy.team_size();
+    auto vector_length = m_policy.impl_vector_length();
+
+    const FunctorType a_functor(m_functor);
+    value_type tmp;
+    ValueInit::init(a_functor, &tmp);
+    static constexpr bool UseReducer = is_reducer_v<ReducerType>;
+
+    if constexpr (!UseReducer) {
+#pragma acc parallel loop gang vector reduction(+ : tmp) copyin(a_functor)
+      for (int i = 0; i < league_size; i++) {
+        int league_id = i;
+        typename Policy::member_type team(league_id, league_size, 1,
+                                          vector_length);
+        a_functor(team, tmp);
+      }
+      m_result_ptr[0] = tmp;
+    } else {
+      OpenACCReducerWrapperTeams<ReducerType, FunctorType, Policy,
+                                 TagType>::reduce(tmp, m_policy, a_functor);
+      m_result_ptr[0] = tmp;
+    }
+  }
+
+  template <class ViewType>
+  inline ParallelReduce(
+      const FunctorType& arg_functor, const Policy& arg_policy,
+      const ViewType& arg_result_view,
+      std::enable_if_t<Kokkos::is_view<ViewType>::value, void*> = nullptr)
```

Comment:
```suggestion
      std::enable_if_t<Kokkos::is_view_v<ViewType>::value>* = nullptr)
```

Reply:
This change is reverted back to the original one, since the definition of is_view_v<> cannot be found.

Reply:
Changed `is_view<>` to `is_view_v<>` after rebasing this branch.